src/utils/sources-tree/getDirectories.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+
+import { createParentMap, nodeHasChildren } from "./utils";
+import { getURL } from "./getURL";
+import type { Node } from "./types";
+
+function findSource(sourceTree: Node, sourceUrl: string): Node {
+  let returnTarget = null;
+  function _traverse(subtree) {
+    if (nodeHasChildren(subtree)) {
+      for (let child of subtree.contents) {
+        _traverse(child);
+      }
+    } else if (!returnTarget) {
+      if (subtree.path.replace(/http(s)?:\//, "") == sourceUrl) {
+        returnTarget = subtree;
+        return;
+      }
+    }
+  }
+
+  sourceTree.contents.forEach(_traverse);
+
+  if (!returnTarget) {
```

Comment:
this is the only logic change. We now return the root if nothing is found so we can promise to have a node returned